src/aics-image-viewer/shared/utils/playControls.ts

```diff
@@ -0,0 +1,80 @@
+import { AxisName } from "../types";
+
+type PlayAxisName = AxisName | "t";
+
+const PLAY_STEP_INTERVAL = 125;
+
+export default class PlayControls {
+  playingAxis: PlayAxisName | null = null;
+  playWaitingForLoad = false;
+  playHolding = false;
+  playTimeoutId = 0;
+
+  public getVolumeIsLoaded?: () => boolean;
+  public stepAxis?: (axis: PlayAxisName) => void;
+  public onPlayingAxisChanged?: (axis: PlayAxisName | null) => void;
+
+  private setPlayingAxis(axis: PlayAxisName | null): void {
+    this.playingAxis = axis;
+    this.onPlayingAxisChanged?.(axis);
+  }
+
+  private playStep(): void {
+    if (!this.playingAxis || this.playHolding || !this.stepAxis) {
+      return;
+    }
+    // If the volume is not loaded, wait for it to load before continuing
+    if (!this.getVolumeIsLoaded?.()) {
+      this.playWaitingForLoad = true;
+      return;
+    }
+
+    this.stepAxis(this.playingAxis);
+    this.playTimeoutId = window.setTimeout(this.playStep.bind(this), PLAY_STEP_INTERVAL);
```

Comment:
curiosity question, what happens if you don't bind `playStep` here?

Reply:
Without `bind`, JavaScript helpfully binds the window object to `this` within `playStep`... so since the first thing `playStep` does is null-check some properties, and since those properties don't exist on `window`, playback just stops after one frame.

Reply:
Ah, thank you, JavaScript!

(thanks for the explanation!)

Reply:
An equivalent alternative would be to make a simple arrow function, right?  setTimeout(()=>this.playStep()...